tests/selenium/test_sizing.py

```diff
@@ -0,0 +1,270 @@
+    table_r0c1_cells = table.find_elements_by_css_selector(
+        ".cell-0-1 > table > tbody > tr:last-of-type > *"
+    )
+    table_r1c0_cells = table.find_elements_by_css_selector(
+        ".cell-1-0 > table > tbody > tr:last-of-type > *"
+    )
+    table_r1c1_cells = table.find_elements_by_css_selector(
+        ".cell-1-1 > table > tbody > tr:last-of-type > *"
+    )
+
+    # this test is very dependent on the table's implementation details..  we are testing that all the cells are
+    # the same width after all..
+
+    # make sure the r1c1 fragment contains all the cells
+    assert len(target_cells) == len(table_r1c1_cells)
+
+    # for each cell of each fragment, allow a difference of up to 1px either way since
+    # the resize algorithm can be off by 1px for cycles
+    for i, target_cell in enumerate(target_cells):
+        assert abs(target_cell.size["width"] - table_r1c1_cells[i].size["width"]) <= 1
+
+    if len(table_r0c0_cells) != 0:
+        assert abs(target_cell.size["width"] - table_r0c0_cells[i].size["width"]) <= 1
+
+    if len(table_r0c1_cells) != 0:
+        assert abs(target_cell.size["width"] - table_r0c1_cells[i].size["width"]) <= 1
+
+    if len(table_r1c0_cells) != 0:
+        assert abs(target_cell.size["width"] - table_r1c0_cells[i].size["width"]) <= 1
+
+
+def test_szng001_widths_on_style_change(test):
+    base_props = dict(
+        data=[
+            {"a": 85, "b": 601, "c": 891},
+            {"a": 967, "b": 189, "c": 514},
+            {"a": 398, "b": 262, "c": 743},
+            {"a": 89, "b": 560, "c": 582},
+            {"a": 809, "b": 591, "c": 511},
+        ],
+        columns=[
+            {"id": "a", "name": "A"},
+            {"id": "b", "name": "B"},
+            {"id": "c", "name": "C"},
+        ],
+        style_data={
+            "width": 100,
+            "minWidth": 100,
+            "maxWidth": 100,
+            "border": "1px solid blue",
+        },
+        row_selectable="single",
+        row_deletable=True,
+    )
+
+    styles = [
+        dict(
+            style_table=dict(
+                width=500, minWidth=500, maxWidth=500, paddingBottom=10, display="none"
+            )
+        ),
+    ]
+
+    fixes = [
+        dict(),
+        dict(fixed_columns=dict(headers=True)),
+        dict(fixed_rows=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True), fixed_rows=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True, data=1)),
+        dict(fixed_rows=dict(headers=True, data=1)),
+        dict(
+            fixed_columns=dict(headers=True, data=1),
+            fixed_rows=dict(headers=True, data=1),
+        ),
+    ]
+
+    variations = []
+    i = 0
+    for style in styles:
+        for fix in fixes:
+            variations.append({**style, **fix, **base_props, "id": "table{}".format(i)})
+            i = i + 1
+
+    variations_range = range(0, len(variations))
+
+    tables = [DataTable(**variation) for i, variation in enumerate(variations)]
```

Comment:
```suggestion
    tables = [DataTable(**variation) for variation in variations]
```

Reply:
https://github.com/plotly/dash-table/pull/793/commits/4dcc753d2ba65cec1dd14e51cc3e59b195e79f70 and https://github.com/plotly/dash-table/pull/793/commits/575aa2c4bc838474bcfa251c8596897be6c76d17#diff-cd5fbde6c058807ca1b16f17f7daa4d8R110